### ISSUE TITLE
EC2 profile/role argmuent for iterative task

### ISSUE THREAD

**Issue by mattlbeck** [iterative/terraform-provider-iterative#501]
Ability to assign an EC2 role to iterative_task would be the most straightforward way of allowing interaction with other AWS services via the AWS CLI, e.g. read/write to S3.

**Comment by 0x2b3bfa0** [comment]
Action point: port https://github.com/iterative/terraform-provider-iterative/pull/234 to `iterative_task`

**Comment by dacbd** [comment]
Do you have feelings on what the hcl for that should look like? A block, like storage? a single string like machine/runner?

**Comment by 0x2b3bfa0** [comment]
Usually, “permissions sets” (e.g. instance profile, service account identifier, ...) are passed as a single string reference,[^1] so it's fine to use a string.

[^1]: If I understand the scope correctly, we want to allow users to specify existing “permission sets” and not create new ones as part of the task.

**Comment by dacbd** [comment]
Agreed, we should _**not**_ manage the "permission" set for them. It should be predefined elsewhere so that several different or parallel tasks can reference the same set.

For more "bureaucratic" 📚  setups it would be frowned upon to use a tool that constantly creates/deletes roles/service accounts. Besides, I know a few places that limit the cloud credentials of their data scientists 😉

**Comment by dacbd** [comment]
I mostly bring up the option to try it as map/block because a single string might not translate well to k8s, and I don't like how like the GCP version is `service_account@project-name.iam.gserviceaccount.com,scropes=thing1,thing2,thing3` 😑

**Comment by dacbd** [comment]
@iterative/cml are we fine with this being configured by a single string?

**Comment by 0x2b3bfa0** [comment]
If the goal is referencing an existing role, I think it's the best approach.

**Comment by casperdcl** [comment]
single string seems fine, unless there are specific examples which are only feasible with `list(str)`
